docs/locals-generation.md

```diff
@@ -0,0 +1,188 @@
+# Locals Generation
+
+Terramate provides a way to integrate its [globals](globals.md) and
+[metadata](metadata.md) on Terraform code by exporting then as
+[Terraform locals](https://www.terraform.io/language/values/locals) that
+then can be referenced by any Terraform code inside the stack.
+
+To make use of [globals](globals.md) and [metadata](metadata.md) define
+a **export_as_locals** block on a [Terramate configuration file](config.md)
+defining all locals that you want exported.
+
+Given these globals defined somewhere inside the project:
+
+```hcl
+globals {
+  data      = "data"
+  more_data = "more data"
+}
+```
+
+To use then directly on Terraform you can instruct code generation
+to create them as locals:
+
+```hcl
+export_as_locals {
+  data      = globals.data
+  more_data = globals.more_data
+}
+```
+
+Exporting using multiple blocks on the same [configuration file](config.md)
+is also allowed, as far as no local is redefined, the previous example could
+also be written like this:
+
+```hcl
+export_as_locals {
+  data      = globals.data
+}
+
+export_as_locals {
+  more_data = globals.more_data
+}
+```
+
+Exported locals can be defined on any [configuration file](config.md)
+so you can have core/base export locals definitions that will generate
+locals for all stacks that are a sub directory of the [configuration](config.md).
+
+Exported locals, just as [globals](globals.md), are evaluated on the context
+of a stack, evaluation starts with exported locals defined on the stack itself
+and then keeps going up on the file system until the project root is reached.
+
+If exported locals don't have the same name, then they are just merged
+together as they are evaluated going up on the project file system.
+
+Exported locals can be overridden, just as [globals](globals.md), the most
+specific exported locals override the more general ones, where by specific
+we mean the definition closest to the stack being evaluated.
+
+Given a project structured like this:
+
+```
+.
+└── stacks
+    ├── stack-1
+    │   └── terramate.tm.hcl
+    └── stack-2
+        └── terramate.tm.hcl
+```
+
+Given a set of globals at the project root **./terramate.tm.hcl**:
+
+```hcl
+globals {
+  data          = "data"
+  more_data     = "more data"
+  yet_more_data = "YMD"
+}
+```
+
+If we define a [configuration file](config.md) for all stacks 
+that export some locals at **stacks/terramate.tm.hcl**:
+
+```hcl
+export_as_locals {
+  data      = globals.data
+  more_data = globals.more_data
```

Comment:
```suggestion
  data      = global.data
  more_data = global.more_data
```

Reply:
ow man I do this shit all the time x_x (switching global to globals :laughing: ). thanks for catching it